Content.Server/Power/Generator/GeneratorSystem.cs

```diff
@@ -198,6 +200,8 @@ public void SetFuelGeneratorOn(EntityUid uid, bool on, FuelGeneratorComponent? g
 
         generator.On = on;
         UpdateState(uid, generator);
+        if (HasComp<EnergyDomeGeneratorComponent>(uid)) // Starlight
+            _energyDomeSystem.Toggle(uid, on); // Starlight
```

Comment:
It feels a bit weird for GeneratorSystem to care about EnergyDomeGeneratorComponent. Could we set it up with bluespace-foo as a new Voltage type (without a defined cable), and just have this be a PowerSupplier and PowerConsumer, and have it toggle on and off when energy is available like other passive machines?